### ISSUE TITLE
cursor jumps to end of bibentry code in entry editor

### ISSUE THREAD

**Issue by ilippert** [JabRef/jabref#8173]
### JabRef version

Latest development branch build (please note build date below)

### Operating system

GNU / Linux

### Details on version and operating system

JabRef 5.4--2021-10-19--f328363 Linux 5.14.13-200.fc34.x86_64 amd64  Java 16.0.2  JavaFX 17.0.0.1+1

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. edit bibtex code in entry editor, cursor in the midst of the entry somewhere
2. switch to another software 
3. switch back to Jabref: Cursor is at the end of the bibentry code

This differs to https://github.com/JabRef/jabref/issues/5904#

**Comment by Siedlerchr** [comment]
First of all: This issue is very old. Does this still happen? Reproduce first.

**Comment by RishabhRawat9** [comment]
Yes it is still there I did reproduce it, should I share a gif showing this bug?

**Comment by RishabhRawat9** [comment]
https://github.com/user-attachments/assets/fb408975-e840-4f82-9767-d26dabe85be4
Hello @Siedlerchr , you can see how after making an edit in the code if you switch to another app and come back the cursor jumps to the end.

**Comment by Siedlerchr** [comment]
okay thanks for the reproducer video! 
You can continue working on this, but please read through the PR description/format etc

**Comment by RishabhRawat9** [comment]
Hi @Siedlerchr, while testing this issue I came across another problem.

Steps to reproduce:
1. Open the entry editor for an entry.
2. Go to the source tab and make an edit (or place the cursor there).
3. Click on another entry of the same type.

The previously selected entry's source is then written into the newly selected entry (overwriting the previous entry)
Should I open a new issue for this, or is this already a known issue? I couldn't find an existing report.
Should I continue with the cursor issue?

Here's a video demonstrating the behavior:
https://github.com/user-attachments/assets/dfaf39c3-2cc0-403b-9477-760cd9a4dfed

**Comment by Siedlerchr** [comment]
@RishabhRawat9 You can investigate this as well. I could also reproduce it, but only sometimes. I think it happens when you don't really click on the other entry row but a bit above/next to it so that it's not blue
